src/not_fact_samplers.jl

```diff
@@ -150,34 +147,55 @@ function pdmp(∇ϕ!, t0, x0, θ0, T, c::Bound, Flow::Union{BouncyParticle, Boom
 end
 
 
-##################################
+##################################    ##################################
+function ab(t, x, θ, C::LocalBound, vdϕ::Number, v, B::BouncyParticle)
+    @assert vdϕ isa Number
+    (C.c + vdϕ, v, t + 2sqrt(length(θ))/C.c/norm(θ, 2))
+end
 
-function pdmp_inner!(rng, dϕ::F1, ∇ϕ!::F2, ∇ϕx, t, x, θ, c::Bound, abc, (t′, renew), τref, (acc, num),
-    Flow::BouncyParticle, args...; subsample=false, oscn=false, factor=1.5, adapt=false) where {F1, F2}
+function next_event1(rng, u::Tuple, abc, flow)
+    t, x, v, V = u
+    a, b, Δ = abc
+    τ = t + poisson_time(a, b, rand(rng))
+    τrefresh = t + waiting_time_ref(rng, flow)/V
+    when, what = findmin((τ, Δ, τrefresh))
+    return when, (:bounce, :expire, :refresh)[what]
+end
+
+function pdmp_inner!(rng, dϕ::F1, ∇ϕ!::F2, ∇ϕx, t, x, θ, V, c::Bound, abc, (t′, action), Δrec, (acc, num),
+    flow::BouncyParticle, args...; subsample=false, oscn=false, factor=1.5, adapt=false) where {F1, F2}
     while true
-        if τref < t′
-            t, _ = move_forward!(τref - t, t, x, θ, Flow)
-            refresh!(rng, θ, Flow)
+        if t + Δrec/V <= t′ # record! (large speed, more records)
```

Comment:
Take samples of the particle position at deterministic time intervals (to reduce sampling error), but measure time proportional to speed (large speed, more samples taken)